kolibri/core/assets/src/core-app/mediator.js

```diff
@@ -321,4 +336,52 @@ module.exports = class Mediator {
   off(...args) {
     this._eventDispatcher.$off(...args);
   }
+
+  _fetchLanguageAssets(moduleName, language) {
+    let promise;
+    if (this._languageAssetRegistry[moduleName] &&
+      this._languageAssetRegistry[moduleName][language] &&
+      this._languageAssetRegistry[moduleName][language].promise) {
+      promise = this._languageAssetRegistry[moduleName][language].promise;
+    } else {
+      promise = new Promise((resolve, reject) => {
```

Comment:
Heh, this is where my JS illiteracy will show. What's a promise and how is it better than a plain callback? Callbacks aren't meant to be broken y'know.